app/code/core/Mage/Core/Model/App.php

```diff
@@ -384,15 +384,10 @@ public function run($params)
      */
     protected function _initEnvironment()
     {
-        if (Mage::getIsDeveloperMode() && class_exists('\Spatie\Ignition\Ignition')) {
-            \Spatie\Ignition\Ignition::make()
-                ->applicationPath(Mage::getBaseDir())
-                ->register();
-        } else {
-            $this->setErrorHandler(self::DEFAULT_ERROR_HANDLER);
-        }
-
+        Mage::dispatchEvent('core_app_init_before', ['app' => $this]);
+        $this->setErrorHandler(self::DEFAULT_ERROR_HANDLER);
```

Comment:
are you sure you don't want to invert these two? since you set another error handled in the module?

Reply:
good point
we can also add 2 events here _before and _after, then observe after event
and maybe better to name it more specific `core_app_initenv_*` cause we have more _init methods (_initCache(),  _initBaseConfig() etc..)
what do you think?

Reply:
interesting topic, i'm not sure at the moment, from one side I wouldn't add many events (especially for a method that's called every time) from the other side they would be helpful in some cases, I'll think about it